File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        TreeSet<Integer> set = new TreeSet<>();
        for (int num: nums)
            set.add(num);
        
        answer = Math.min(set.size(), nums.length/2);
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    
    let mySet = new Set();
    for (let i = 0; i < nums.length; i++) {
        mySet.add(nums[i]);
    
        if (mySet.size == nums.length/2)
            return mySet.size;
    }
    
    return mySet.size;
}